app/models/concerns/edm/entity/referenced_records.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module EDM
+  module Entity
+    module ReferencedRecords
```

Comment:
I'm concerned about how tight a coupling this introduces to the Blacklight interface to the Record API, *within* the entity models.

While it would be nice to have model-level associations to retrieve CHOs etc from the Record API, I do not think it can be justified at the cost of bringing Blacklight documents, config and controller instances into these models.

I'd suggest therefore that retrieval of referenced records be moved up into `EntitiesController` which then need only use `#search_results` to construct `@referenced_records` as it already has the Blacklight config, and is a controller.